src/validation/merkle.rs

```diff
@@ -154,10 +154,10 @@ fn tree_with_two_layers_should_reach_all_nodes() {
 #[test]
 fn four_blocks_should_built_seven_nodes() {
     let blocks = [
-        Block::builder(Vec::new()).build(),
-        Block::builder(Vec::new()).build(),
-        Block::builder(Vec::new()).build(),
-        Block::builder(Vec::new()).build(),
+        &Block::builder(Vec::new()).build(),
```

Comment:
`(0..4).map(|_| &Block::builder(Vec::new()).build()) ... ` will be more concise.

Reply:
![image](https://user-images.githubusercontent.com/5635032/77671779-83361980-6fa1-11ea-9d2c-7537470dfee0.png)
How to be with temporary references?

Reply:
It can be done like this:
```rust
#[test]
    fn four_blocks_should_built_seven_nodes() {
        let blocks: Vec<_> = (0..4).map(|_| Block::builder(Vec::new()).build()).collect();
        let merkle_tree = MerkleTree::build(&blocks.iter().collect::<Vec<_>>());
        assert_eq!(7, merkle_tree.into_iter().count());
    }
```

But, yes I agree, not very pretty. Still if we have many elements there, it would make sense.

Reply:
@eadventurous sorry - I think that your variant has more cognitive complexity than simple 4 lines of equal code( It's harder to read and write.